util/imagetools/create.go

```diff
@@ -138,12 +138,25 @@ func (r *Resolver) Combine(ctx context.Context, srcs []*Source) ([]byte, ocispec
 		mt = ocispec.MediaTypeImageIndex
 	}
 
+	// annotations are only allowed on OCI indexes
+	annotations := make(map[string]string)
+	if mt == ocispec.MediaTypeImageIndex {
+		for k, v := range ann {
+			if !strings.HasPrefix(k, "index:") {
+				return nil, ocispec.Descriptor{}, errors.Errorf("invalid annotation key %q, expected it to have 'index:' prefix", k)
+			}
+			k = strings.TrimPrefix(k, "index:")
+			annotations[k] = v
```

Comment:
We should also support `manifest-descriptor` here - this would need to modify `newDescs`, but shouldn't be too tricky?

We don't have to do `manifest` right now, since that involves mutating the manifest, so we can leave that as a future extension.

Reply:
Done.